pytorch_lightning/plugins/precision/precision_plugin.py

```diff
@@ -242,3 +242,20 @@ def teardown(self) -> None:
 
         It is the right place to release memory and free other resources.
         """
+
+    def state_dict(self) -> Dict[str, Any]:
```

Comment:
Did you consider making the class use `Stateful` as a mixin?

Reply:
just was thinking the same as I see it already a few times...

Reply:
originally proposed `Protocol` so that users could simply implement `state_dict/load_state_dict` without having to extend anything. but now that we're adding them to lightning base classes, I think mixin could be better - can change this in another PR